src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -685,8 +687,9 @@ def build_vmss_resource(cmd, name, naming_prefix, location, tags, overprovision,
         storage_properties['imageReference'] = {
             'id': image
         }
-
-    storage_properties['dataDisks'] = [v for k, v in disk_info.items() if k != 'os']
+    data_disks = [v for k, v in disk_info.items() if k != 'os']
+    if data_disks:
```

Comment:
It appears vmss doesn't accept an empty array when with unmanaged disks; hence the extra condition  here